assemble/bin/accumulo-cluster

```diff
@@ -129,9 +129,21 @@ function control_service() {
     [[ "$service" == "compactor" ]] && ACCUMULO_SERVICE_INSTANCE="${inst_id}_${5}"
 
     if [[ $host == localhost || $host == "$(hostname -s)" || $host == "$(hostname -f)" || $host == $(get_ip) ]] ; then
-      ACCUMULO_SERVICE_INSTANCE="${ACCUMULO_SERVICE_INSTANCE}" "${bin}/accumulo-service" "$service" "$control_cmd" "${@:4}"
+      #
+      # The server processes take arguments (e.g. -a, -p, -o, -q [in the case of the Compactor]).
+      #
+      if [[ $# -gt 3 ]]; then
+        ACCUMULO_SERVICE_INSTANCE="${ACCUMULO_SERVICE_INSTANCE}" "${bin}/accumulo-service" "$service" "$control_cmd" "${@:4}"
+      else
+        ACCUMULO_SERVICE_INSTANCE="${ACCUMULO_SERVICE_INSTANCE}" "${bin}/accumulo-service" "$service" "$control_cmd"
+      fi
     else
-      $SSH "$host" "bash -c 'ACCUMULO_SERVICE_INSTANCE=${ACCUMULO_SERVICE_INSTANCE} ${bin}/accumulo-service \"$service\" \"$control_cmd\" \"${@:4}\" '"
+      if [[ $# -gt 3 ]]; then
+        EXTRA_ARGS="${@:4}"
+        $SSH "$host" "bash -c 'ACCUMULO_SERVICE_INSTANCE=${ACCUMULO_SERVICE_INSTANCE} ${bin}/accumulo-service \"$service\" \"$control_cmd\" $EXTRA_ARGS '"
```

Comment:
This is flattening the array into a single string, which can mess with our ability to reason about the quoting. It's probably better to be consistent, and use the array form (or leave the original, which was already an array):
```suggestion
        EXTRA_ARGS=("${@:4}")
        $SSH "$host" "bash -c 'ACCUMULO_SERVICE_INSTANCE=${ACCUMULO_SERVICE_INSTANCE} ${bin}/accumulo-service \"$service\" \"$control_cmd\" \"${EXTRA_ARGS[@]}\" '"
```

Reply:
I can test this out and see if it works.

Reply:
So, the args come over as one string in the way that you suggested and does not work. Error in the compactor.err file:

```
Was passed main parameter '-q q2' but no main parameter was defined in your arg class
```

Reply:
Okay, that's fine then. It looks like it's flattening them already on the local end. So, I guess we need to pass them unquoted so they get treated as separate args... or we could do something crazy like build the command using an array, but that seems like overkill. I think your solution that worked without the quoting is fine.